acstools/focus_diverse_epsfs.py

```diff
@@ -0,0 +1,452 @@
+        LOG.error("Invalid download location")
+        return []
+
+    if isinstance(input_list, str):  # Text file with one rootname per line
+        orig_lines = Path(input_list).read_text().rsplit()
+    elif not isinstance(input_list, (list, tuple)):
+        LOG.error("Invalid input list")
+        return []
+    else:  # List of rootnames
+        orig_lines = input_list
+
+    # Should throw out things that are obviously wrong here to minimize network hit.
+    lines = [line for line in orig_lines if _validate_acs_ipsoot(line)]
+    n_lines = len(lines)
+
+    if n_lines < 1:  # Nothing to do
+        return []
+
+    if n_lines == 1:  # No need for multiprocessing
+        return [psf_retriever(lines[0], download_location)]
+
+    if n_lines < num_workers:  # No need that many workers
+        num_workers = n_lines
+
+    # perform multiprocessing with dask
+    dask_results = [dask.delayed(psf_retriever)(line, download_location) for line in lines]
+
+    # run with progress bar
+    with ProgressBar():
+        results = dask.compute(
+            *dask_results, num_workers=num_workers, scheduler='processes')
+
+    return results
+
+
+def interp_epsf(ePSFs, x, y, chip, pixel_space=False, subpixel_x=0, subpixel_y=0):
+    """Function to perform further spatial interpolations given the input ePSF array.
+    It uses bi-linear interpolation for the
+    integer pixel shifts, and bi-cubic interpolation for any specified subpixel phase shifts.
+
+    This function allows users to interpolate the provided ePSF arrays
+    to any arbitrary ``(x, y)`` coordinates. It can be called with ``pixel_space=True`` to
+    downsample the ePSF into detector space. Subpixel phase shifts can be applied by
+    setting ``subpixel_x`` and ``subpixel_y`` between 0 and 0.99. Note that a 1 pixel border
+    is removed from the subpixel phase shifted ePSF, such that the final dimensions are 23x23.
+    Results from this subpixel phase shift routine may differ from other algorithmic
+    implementations, typically at the level of <0.5% in the core of the ePSF.
+
+    .. note::
+
+        This function requires an optional ``scipy`` dependency to be installed
+        for ``pixel_space=True``.
+
+    .. note::
+
+        This function requires users to specify the WFC chip (WFC1 or WFC2).
+
+    Parameters
+    ----------
+    ePSFs : numpy.ndarray
+        Array with the ePSFs.
+
+    x : int
+        X-coordinate (0-indexed) of the desired output ePSF.
+
+    y : int
+        Y-coordinate (0-indexed) of the desired output ePSF. Please note that the range here is between
+        0 and 2048, inclusive.
+
+    chip : str
+        String corresponding to which ACS/WFC detector the user is specifying the coordinates on,
+        either "WFC1" or "WFC2".
+
+    pixel_space : bool
+        If `True`, downsample the ePSF into detector space.
+
+    subpixel_x, subpixel_y : float
+        The desired subpixel coordinate, between 0 and 0.99.
```

Comment:
@gsanand , like this? I cannot say I understand the last sentence but I am also not an expert in this field.

```suggestion
        The desired subpixel coordinate, between 0 and 0.99.
        ``pixel_space`` must be set to `True` to use this option.
        The defaults for both x and y are 0, and correspond to the star being centered within the individual pixel.
```